### PR TITLE
ROC-3026 Save defendant response in ccd

### PR DESCRIPTION
**Pull request by dharmendrak**
https://tools.hmcts.net/jira/browse/ROC-3026

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java [hunk 1]
```diff
@@ -184,9 +184,15 @@ public void saveCountyCourtJudgment(String authorisation, Claim claim, CountyCou
...
 
-    public void saveDefendantResponse(long claimId, String defendantId, String defendantEmail, Response response) {
+    public void saveDefendantResponse(
+        Claim claim,
+        String defendantId,
+        String defendantEmail,
+        Response response,
+        String authorization
+    ) {
         // When this is saved in CCD ensure a Forbidden response is returned to the client if they
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java]: can you test this and remove the comment if true? it should work after I mapped the feign exception

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java [hunk 2]
```diff
@@ -79,6 +86,22 @@ public CaseDetails saveCountyCourtJudgment(
...
+    public CaseDetails saveDefendantResponse(
+        String authorisation,
+        Claim claim,
+        String defendantEmail,
+        Response response,
+        String defendantId
+    ) {
+
+        CCDCase ccdCase = this.caseMapper.to(claim);
+        ccdCase.setSubmitterId(defendantId);
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: isn't this claimant id?

Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: That does not look right. By submitted ID is changing?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java [hunk 3]
```diff
@@ -79,6 +86,22 @@ public CaseDetails saveCountyCourtJudgment(
...
+        ccdCase.setResponse(responseMapper.to((FullDefenceResponse) response));
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: do we need to type cast to `FullDefenceResponse` ?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java [hunk 4]
```diff
@@ -79,6 +86,22 @@ public CaseDetails saveCountyCourtJudgment(
         return this.update(authorisation, ccdCase, DEFAULT_CCJ_REQUESTED);
     }
 
+    public CaseDetails saveDefendantResponse(
```
Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: Can you please make argument list consistent with other places. We should have the following order.

```
String defendantId,
String defendantEmail,
Response response,
```

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDCase.java [hunk 5]
```diff
@@ -21,5 +22,7 @@
     private CCDClaim claimData;
     private CCDCountyCourtJudgment countyCourtJudgment;
     private String countyCourtJudgmentRequestedAt;
-
+    private String defendantEmail;
+    private CCDResponse response;
+    private String respondedAt;
```
Comment (sabahirfan) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDCase.java]: You need to add defendantId here as well.

Comment (timja-kainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDCase.java]: no defendant id should not be here!!!!

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java [hunk 6]
```diff
@@ -83,29 +86,28 @@ public CaseDetails saveCountyCourtJudgment(
...
-        ccdCase.setSubmitterId(defendantId);
         ccdCase.setResponse(responseMapper.to((FullDefenceResponse) response));
         ccdCase.setDefendantEmail(defendantEmail);
         ccdCase.setRespondedAt(now().format(ISO_DATE_TIME));
-        return this.update(authorisation, ccdCase, DEFENCE_SUBMITTED);
+        return this.update(authorisation, ccdCase, DEFENCE_SUBMITTED, defendantId);
     }
 
-    public CaseDetails update(String authorisation, CCDCase ccdCase, CaseEvent caseEvent) {
+    public CaseDetails update(String authorisation, CCDCase ccdCase, CaseEvent caseEvent, String userId) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: Don't need to pass userId as userId can be extracted from `authorisation` here like ```String userId = userService.getUserDetails(authorisation).getId();```

**Pull request merged**